app/off.py

```diff
@@ -10,15 +11,20 @@
 from .wikidata_utils import get_wikidata_entity, image_thumbnail, wikidata_props
 
 
+@alru_cache()
+async def global_quality(source_url):
+    async with aiohttp.ClientSession() as session:
+        quality_url = f"{source_url}/data-quality.json"
+        async with session.get(quality_url) as resp:
+            return await resp.json()
+
+
 async def data_quality(url, path):
     """
     Helper function to return issues for data-quality
     """
-    async with aiohttp.ClientSession() as session:
-        source_url = urljoin(url, path)
-        quality_url = f"{source_url}/data-quality.json"
-        async with session.get(quality_url) as resp:
-            data = await resp.json()
+    source_url = urljoin(url, path)
+    data = await global_quality(source_url)
```

Comment:
the idea was to use the cache only for facets that takes a lot of time to compute, mainly, here, when it's data-quality/.

A simple way to do this could be to simply count the number of "/" in path. or even test if we are requesting path="data-quality/".

You may have two versions of fetch_quality like that:
```python

async def fetch_quality(source_url):
    async with aiohttp.ClientSession() as session:
        quality_url = f"{source_url}/data-quality.json"
        async with session.get(quality_url) as resp:
            return await resp.json()

# cached version of fetch_quality for slow requests
global_quality = alru_cache(fetch_quality)
```

(a decorator is just a function you call on a function)

Reply:
@alexgarel  How I can put "source_url" in "global_quality" ?

Reply:
because it seems we can't put function value in "alru_cache"
```
facets-knowledge-panels-facets-api-1  |     raise NotImplementedError("{} decorating is not supported".format(fn))
facets-knowledge-panels-facets-api-1  | NotImplementedError: <coroutine object fetch_quality at 0x7fc67fb77840> decorating is not supported
```

Reply:
global_quality is a function with same signature as fetch_quality.

The first time we call it, it will behave as fetch_quality, but the second time, it will use its cache.